tests/integration/input/test_number_input.py

```diff
@@ -5,16 +5,8 @@
 
 def test_inni001_invalid_numbers(ninput_app, dash_dcc):
     dash_dcc.start_server(ninput_app)
-    for invalid_number in (
-        "10e10000",
-        "e+++eeeeeE-",
-        "120.2.33",
-        "12-.3",
-        "-3.3.3",
-        "0..0",
-    ):
+    for invalid_number in ("10e10000", "e+++eeeeeE-", "12-.3"):
```

Comment:
120.2.33
-3.3.3
0..0

In Chrome 87 the above entries get processed into valid numbers as the invalid characters are simply removed

Reply:
120.2.33 -> 120.233
-3.3.3 -> -3.33
0..0 -> 0.0

Reply:
Oh interesting - what "removed" means is those characters are not accepted into the input. You type them and they just don't appear.

Begs the question why they still allow you to type `10e10000` - if they know to refuse a second decimal point, why don't they also refuse another zero past `10e100`? There's no way to continue typing and turn that into a valid number. Likewise `12-.3` makes no sense at all. Really the only non-numbers they _need_ to allow you to type are incomplete numbers, so I'd probably just put a few of those in and take out these others. Like perhaps `-.`, `12e-`. Otherwise we'll just see this problem again when they tighten up their logic further.